lib/utils.js

```diff
@@ -81,4 +81,35 @@ class NullLogger {
 }
 utils.nullLogger = new NullLogger();
 
+/**
+ * Deep merge two objects.
+ * @param {Object} target
+ * @param {Object} ...sources
+ * @return {Object}
+ */
+utils.mergeDeep = (target, ...sources) => {
+    if (!sources.length) {
```

Comment:
The stop condition should include `!sources`, `!target` and `!isObject(target)` too.

Reply:
Done.